src/gwt/src/org/rstudio/core/client/widget/AriaLiveStatusWidget.java

```diff
@@ -63,8 +66,23 @@ public void clearMessage()
       public void run()
       {
          getElement().setInnerText(resultsMessage_);
+         if (clearReaderTimer_.isRunning())
+            clearReaderTimer_.cancel();
+         clearReaderTimer_.schedule(4000);
       }
    };
 
+   /**
+    * Timer for clearing the previous message if nothing new arrives
+    */
+   private Timer clearReaderTimer_ = new Timer()
+   {
+     @Override
+     public void run()
+     {
+        getElement().setInnerText("");
+     }
+   };
+
```

Comment:
This seems to just be an extension of what's already there but I always get a little nervous when multiple timers start being about. This looks sound but I always ask in this situation "are there any funny conditions we could get in?"

There are some hardcoded times (like the 4000ms on line 71) as well that, at least in the next revision, might want to get pulled out to at least class constants. Not a big deal for this scope but probably something to think about as these things continue to grow in complexity.

If we end up with a handful of timers in flight in the future it may be worth offloading some of these status changes to an event bus system.

Reply:
All fair points. I hate when I add (or see) timers in the code, but if you look, we have dozens of them, and most have hardcoded values. That doesn't make it a "good thing".

In this case, there are definitely "funny conditions we could get in", and they are unknowable without testing combinations of screen readers, browsers, user controlled settings for screen reader speech rate, and length of the messages being injected. Our code has no idea how long it will take for a given message to be read, or if a screen reader will read the message even after the text has been cleared.

Two reasons I'm doing this clear after a delay:

1) The text in a live-alert region, although invisible to sighted users, is visible to screen readers so if the user is navigating through the UI with the screen reader cursor, they will come across it and it is weird to get whatever the last announcement was (which might have happened hours / days ago) read again. So I want to clear it after a while to reduce the chances of that.

2) If you use the new command I added that reads the current editor status (row, column, etc), it is quite reasonable that a user might issue that command again while on the same line (missed details the first time). A screen reader will not read the same announcement twice in a row, so clearing it increases the chances of them hearing it the second time (but only if they wait 4 seconds, so far from perfect). Another strategy would be to have two live regions, and have the code alternate between them with each announcement, clearing the other one when updating the current one. Would still want the timer to clear each region after some amount of time, but would be more robust in general, I'll open a task to look into that in the future.